client/my-sites/site-settings/theme-setup/index.jsx

```diff
@@ -14,41 +14,71 @@ import QueryActiveTheme from 'components/data/query-active-theme';
 import QueryTheme from 'components/data/query-theme';
 import ThemeSetupCard from './theme-setup-card';
 import ThemeSetupPlaceholder from './theme-setup-placeholder';
-import { getSelectedSite } from 'state/ui/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
 import { getActiveTheme, getTheme } from 'state/themes/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
 import { toggleDialog } from 'state/ui/theme-setup/actions';
 
-let ThemeSetup = ( { site, themeId, theme, translate, activeSiteDomain, toggleDialog } ) => {
-	const onBack = () => {
-		page( '/settings/general/' + activeSiteDomain );
+class ThemeSetup extends Component {
+	componentDidUpdate( prevProps ) {
+		const {
+			siteIsJetpack,
+			siteId,
+		} = this.props;
+
+		if ( siteId !== prevProps.siteId && siteIsJetpack ) {
+			this.redirectToGeneral();
+		}
+	}
+
+	redirectToGeneral = () => {
+		const { siteSlug } = this.props;
+		page.redirect( '/settings/general/' + siteSlug );
```

Comment:
Could we get here without a `siteSlug`? It may be worth adding a default so we don't end up at `/settings/general/undefined`.

```js
const { siteSlug = '' } = this.props;
```